api/src/main/java/com/codeforcommunity/dto/report/UpsertReportWithLibrary.java

```diff
@@ -170,23 +175,24 @@ public void setParentSupport(String parentSupport) {
       fields.add("hasSufficientTraining");
     }
 
-    List<String> timetableFields = this.validateTimetable();
-    fields.addAll(timetableFields);
-
+    List<String> checkInTimetableFields = this.validateTimetable(checkInTimetable);
+    List<String> checkOutTimetableFields = this.validateTimetable(checkOutTimetable);
+    fields.addAll(checkInTimetableFields);
+    fields.addAll(checkOutTimetableFields);
     return fields;
   }
 
-  private List<String> validateTimetable() {
+  private List<String> validateTimetable(JsonNode timetable) {
```

Comment:
⛏️  not sure because im reviewing on my phone but if this isnt referring to any class fields lets make this static

Reply:
^

Reply:
We can't make it static because the method requires use of the logger class field @rymaju